src/services/TechRecordsService.ts

```diff
@@ -373,6 +418,80 @@ class TechRecordsService {
+      return Promise.reject({statusCode: 400, body: isPayloadValid.error.details});
+    }
+    console.log("Passed Initial validation");
+    return this.getTechRecordsList(techRecord.systemNumber, STATUS.ALL, SEARCHCRITERIA.SYSTEM_NUMBER)
+      .then((data: ITechRecordWrapper[]) => {
+        if(data.length !== 1) {
+          // systemNumber search should return a unique record
+          throw new HTTPError(500, ERRORS.NO_UNIQUE_RECORD);
+        }
+        const uniqueRecord = data[0];
+        if(uniqueRecord.techRecord.filter((techRecords: any) => {
+          return techRecords.statusCode === STATUS.CURRENT || techRecords.statusCode === STATUS.PROVISIONAL;}).length!==0) {
+          throw new HTTPError(400, ERRORS.CURRENT_OR_PROVISIONAL_RECORD_FOUND);
+        }
+        uniqueRecord.techRecord.push(techRecord.techRecord[0]);
+        return uniqueRecord;
+        })
+        .catch((error: any) => {
+          throw new HTTPError(error.statusCode, error.body);
+        });
+  }*/
+  public addProvisionalTechRecord(techRecord: ITechRecordWrapper, msUserDetails: any) {
+    return this.manageAddProvisionalLogic(techRecord, msUserDetails);
+  }
+
+  private manageAddProvisionalLogic(techRecord: ITechRecordWrapper, msUserDetails: any) {
+    return this.addNewProvisionalRecord(techRecord, msUserDetails)
+    .then((data: any) => {
+      return this.techRecordsDAO.updateSingle(data)
+        .then((updatedData: any) => {
+          return this.formatTechRecordItemForResponse(updatedData.Attributes);
+        })
+        .catch((error: any) => {
+          throw new HTTPError(error.statusCode, error.message);
+        });
+    })
+    .catch((error: any) => {
+      throw new HTTPError(error.statusCode, error.body);
+    });
+  }
+  public addNewProvisionalRecord(techRecord: ITechRecordWrapper, msUserDetails: any) {
+    const isPayloadValid = validatePayload(techRecord.techRecord[0]);
+    console.log("Validate Payload: ",isPayloadValid);
+    if (isPayloadValid.error) {
+      return Promise.reject({statusCode: 400, body: isPayloadValid.error.details});
+    }
+    console.log("Passed Initial validation");
+    return this.getTechRecordsList(techRecord.systemNumber, STATUS.ALL, SEARCHCRITERIA.SYSTEM_NUMBER)
+      .then((data: ITechRecordWrapper[]) => {
+        if(data.length !== 1) {
+          // systemNumber search should return a unique record
+          throw new HTTPError(500, ERRORS.NO_UNIQUE_RECORD);
+        }
+        const uniqueRecord = data[0];
+        /*if(uniqueRecord.techRecord.filter((techRecords: any) => {
+          return techRecords.statusCode === STATUS.CURRENT || techRecords.statusCode === STATUS.PROVISIONAL;}).length!==0) {
+          throw new HTTPError(400, ERRORS.CURRENT_OR_PROVISIONAL_RECORD_FOUND);
+        }*/// This is unneeded after my chat with Ketan who said that this will all be handled in the FE
+        techRecord.techRecord[0].lastUpdatedAt = new Date().toISOString();
```

Comment:
should be createdAt, createdByName, createdById, since you're creating new record, not updating an existing one